### PR TITLE
Reland "[libc][NFC] adjust time related implementations""

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
Reverts llvm/llvm-project#91657 and Relands #91485

### CODE REVIEW

File: libc/src/__support/time/clock_gettime.h
Comment (lntue) [inline libc/src/__support/time/clock_gettime.h:1]: The whole `clock_gettime` in `src/__support/time` is still very Linux-specific.  Can we still keep them in `src/time/linux` instead?

**Review by lntue — APPROVED** [review]

**Comment by SchrodingerZhu** [comment]
Passed GPU builds on my machine. Lets see if the bots are happy now.

**Comment by jhuber6** [comment]
> Passed GPU builds on my machine. Lets see if the bots are happy now.

Glad to hear someone else got the GPU build running locally. It's probably easier now that you can have the GPU and CPU builds at the same  time.

**Comment by nickdesaulniers** [comment]
> It's probably easier now that you can have the GPU and CPU builds at the same time.

Sorry, I'm just getting back up to speed after some time off. Can you point me at docs on how to do this, or the PR that added this?

**Pull request merged**